### DISCUSSION TITLE
dynamoose.Instance() and model methods to interact with dynamodb regional instance

### DISCUSSION THREAD

**Discussion by RobinAbrahamThomas** [dynamoose/dynamoose#1557] [Q&A]
**Question modified to add more clarity:**

I was hoping the implementation discussed in this [thread ](https://github.com/dynamoose/dynamoose/issues/1002) is available with dynamoose version 3.x.x,

See below code snippet for a better understanding of what I am trying to achieve with dynamoose. 

```
const dynamoose = require("dynamoose");

const regionList = ["eu-central-1", "ap-south-1"];
const stores = {};

/**
 * Method to initialize regional stores
 */
function initializeRegionalStores() {
    for (const region of regionList) {
        const dynamooseInstance = new dynamoose.Instance();
        dynamooseInstance.aws.ddb.set(new dynamoose.aws.ddb.DynamoDB({
            "region": region
        }));
        dynamooseInstance.Table.defaults.set({ create: true });
        // DYNAMOOSE MODEL CREATED IN NEXT LINE IS STORED FOR LATER USAGE TO ACCESS REGIONAL DB INSTANCES 
        // BUT THERE IS NO OPTION TO LINK THIS MODEL TO THE INSTANCE CREATED FOR THE REGION !!.
        stores[region] = dynamoose.model("Data", new dynamoose.Schema({
            "id": String,
        },
            {
                saveUnknown: true,
                timestamps: true
            }
        )
        );
    }
}


/**
 * Method to store data into different regional dynamodb tables of same schema
 * @param data Obect
 */
function storeDataBasedOnLocation(data) {
    if (!regionList.includes(data.region)) {
        throw new Error("Missing region value");
    }

    await stores[data.region].create(data);

} 
```
Since there seems to be no way to instruct a dynamoose model to use a particular dynamoose instance and dynamoose instance has only `Table`, how can we make sure that the model is executing query against the correct regional DynamoDB instance? .

**Comment by fishcharlie** [comment]
Correct. Dynamoose instances only have `Table`.

**Reply by RobinAbrahamThomas** [reply]
Thanks, Charlie, for the quick reply and the awesome Dynamoose module. I realized my question was not clear enough. I have updated it. Could you please check again

**Reply by fishcharlie** [reply]
@RobinAbrahamThomas https://dynamoosejs.com/guide/Table#new-dynamoosetablename-models-options just create a table instance and pass in an array of your models (in this case it looks to only be 1).
